pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -119,25 +118,15 @@ func loadAssignment(c *EdsCluster) *xdsapi.ClusterLoadAssignment {
 	return c.LoadAssignment
 }
 
-func newEndpoint(address string, port uint32) (*endpoint.LbEndpoint, error) {
-	ipAddr := net.ParseIP(address)
-	if ipAddr == nil {
-		return nil, errors.New("Invalid IP address " + address)
+func newEndpoint(e *model.NetworkEndpoint) (*endpoint.LbEndpoint, error) {
```

Comment:
I'm not sure eds is right here - uds is mostly for 'inbound' - returning inline in cds seems more efficient.

May be ok for now  - but we should change in a future pr.

Reply:
It's a trade off between protocol efficiency and Pilot implementation complexity.  Arguably, we might want to switch all clusters generated by ServiceEntry to static, since the CRD contains the endpoints. I agree it's for a separate PR.

Reply:
Its already static.

Reply:
Actually - I'm a bit confused here. I understand using UDS between envoy and an app on the same  machine, for the 'in' cluster. And that's static.

But how will it work for 'out' and EDS ? 

Can you describe the use case again - do we expect envoy to talk with both tcp and uds ? Will the uds socket be on all machines ?

Reply:
Yes, the UDS socket will need to be on all machines for this to work.

In our specific use case, we don't actually care about whether the workloads can reach the service. We just want the cluster created so that a filter can talk to it. However, disabling the creating of a listener from a ServiceEntry is a bigger change from what's already there.

So, in this implementation you will still get an "outbound" TCP listener associated with the ServiceEntry address/port or host/port. This outbound listener points to a cluster with the UDS endpoint(s).

We don't create "inbound" listeners, since those are created by matching the sidecar IP address with the ServiceInstance. Since the address in this case is a UDS, it will never match any sidecar.